### DISCUSSION TITLE
connection refused

### DISCUSSION THREAD

**Discussion by Claudio-dur** [nextflow-io/nextflow#3912] [Q&A]
Hi! 

I am running a pipeline in nextflow and a weird error starts to appear:

Command error:
  WARNING: Could not lookup the current user's information: user: lookup userid xxxxx: connection refused
  FATAL:   Couldn't determine user account information: user: lookup userid xxxxx: connection refused

 The pipeline runs smoothly until a point where this error above appears. If I resume the task, the simulations move forward and the same error appears in a later point. If I resume again, the process continues as expected and the error appears again in a later point and so on... 

Any idea what could this be? I'm running the pipeline in a tmux environment. So far the error appears when using the tool bowtie2samtools. If it goes further in the pipeline, it might appear for instrain as well. 

Thank you for the support! 
Cheers,
Claudio

**Comment by bentsherman** [comment]
Hi Claudio,

We need more context in order to help you effectively. Please refer to the issue template and provide all of the requested details, like the pipeline you ran, configuration, environment, etc.

**Reply by Claudio-dur** [reply]
Hi Bent, 

Thank you for your reply. I will try to answer with much details as possible.

## Bug report 

### Expected behavior and actual behavior

The nextflow pipeline consists on running  two main tools with small ones in between. The two main tools are bowtie2samtools and instrain. The input are fastq files with sequence information from shotgun metagenomics. The output should be a .tsv file with relative bacterial abundance table. The output of bowtie2samtools is taken by instrain to provide the mentioned .tsv file after some clean-ups. 

So far, when running bowtie2samtools, the pipeline stops when analysing certain samples with the :connection refused error. Nonetheless, when rerunning the pipeline with the -resume argument, the samples that gave the error run through without problems, and the error appears later when analysing another sample. If the -resume argument is used again, the pattern is repeated. 

### Steps to reproduce the problem

1.- First, start a tmux server with the command tmux.

2.- Load the nextflow pipeline module

3.- Run the pipeline with proper arguments. 

### Program output 

.nextflow.log output:

`task: name=bowtie2samtools (16); work-dir=/xx/yy/zz/work/4c/6a9db4f120a4e78b3bb01a99793c5f
  error [nextflow.exception.ProcessFailedException]: Process `bowtie2samtools (16)` terminated with an er
ror exit status (255)
May-03 13:04:15.778 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'bo
wtie2samtools (16)'

Caused by:
  Process `bowtie2samtools (16)` terminated with an error exit status (255)

Command executed:

  bowtie2 -p 8 -x /xx/yy/zz/gtdb_genomes_reps_r207.fasta.bt2 -1 40054_4#95_1.fastq.gz -2 40054_4#95_2.fastq.gz | samtools sort -@ 8 -o 40054_4#95".sorted.bam"
  mapping_rate=$(grep "overall alignment rate" .command.err | awk '{ print $1 }')
  echo 40054_4#95,${mapping_rate} > 40054_4#95_mapping_rate.csv

Command exit status:
  255

Command output:
  (empty)

Command error:
 ESC[33mWARNING:ESC[0m Could not lookup the current user's information: user: lookup userid 21668: connection refused
  ESC[31mFATAL:  ESC[0m Couldn't determine user account information: user: lookup userid 21668: connection refused

Work dir:
  /xx/yy/work/4c/6a9db4f120a4e78b3bb01a99793c5f

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
May-03 13:04:15.783 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `bowtie2samtools (16)` terminated with an error exit status (255)
`

### Environment 

* Nextflow version: nextflow/23.04.1-5866
* Java version: openjdk version "11.0.18"
* Operating system: Ubuntu 18.04 (bionic) (5.4.0-144-generic x86_64)
* Bash version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

### Additional context

So far I have had this problem only when running my pipeline with tmux. So I wonder if it is related to it.

Thank you!

**Comment by biscuit13161** [comment]
I'm not sure if this will help others, but I've just been having this issue too, and I stopped getting the error once I changed the partition on the HPC
